// @cursorify/react/dist/components/CursorifyProvider/Cursorify/Cursorify.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("react");
const useDefaultCursorVisibleEffect_1 = __importDefault(require("./useDefaultCursorVisibleEffect"));
const useMouseMoveEffect_1 = __importDefault(require("./useMouseMoveEffect"));
const useMouseStateEffect_1 = __importDefault(require("./useMouseStateEffect"));
const Cursorify = ({ children }) => {
    const ref = (0, react_1.useRef)(null);
    (0, useMouseMoveEffect_1.default)(ref);
    (0, useDefaultCursorVisibleEffect_1.default)();
    (0, useMouseStateEffect_1.default)();
    return ((0, jsx_runtime_1.jsx)("div", { ref: ref, style: {
            pointerEvents: 'none',
            position: 'fixed',
            top: 0,
            left: 0,
            transform: 'translate(-50%, -50%)',
            opacity: 0,
            zIndex: 9999,
        }, children: children }));
};
exports.default = Cursorify;
